lib/suma/commerce/cart.rb

```diff
@@ -101,11 +101,13 @@ def set_item(product, quantity, timestamp:)
 
   # This seems like a reasonable default...
   DEFAULT_MAX_QUANTITY = 12
+  CLOSED_OFFERING_MAX_QUANTITY = 0
 
   def max_quantity_for(offering_product)
     product = offering_product.product
     inv = product.inventory!
     offering = offering_product.offering
+    return CLOSED_OFFERING_MAX_QUANTITY unless offering.available_at?(Time.now)
```

Comment:
There's a reason we don't use `Time.now` like this elsewhere. It needs to be passed in.

Also I would not put this here. You're mixing two ideas- actual quantity, and availability of an offering. Imagine you had the ability for admins to order a thing, irrespective of whether the offering is closed. The quantity still needs to be accurate.

Instead look at line 81 in `api/commerce.rb`. There's already an `available?` predicate which represents a similar concept, and can be used.